### ISSUE TITLE
Need consistency in terminology for product_name/application_name

### ISSUE THREAD

**Issue by mjgiarlo** [projecthydra-labs/hyrax#179]
<a href="https://github.com/vantuyls"><img src="https://avatars.githubusercontent.com/u/9534576?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [vantuyls](https://github.com/vantuyls)**
_Friday Sep 30, 2016 at 22:46 GMT_
_Originally opened as https://github.com/projecthydra/sufia/issues/2762_

----

in:
sufia/app/helpers/sufia/sufia_helper_behavior.rb

lines 7-9 there is some confusing inconsistency in the terms used to describe the application/product (see below):

def application_name
      t('sufia.product_name', default: super)
    end

It would be less confusing if the terminology was consistent.

**Comment by mjgiarlo** [comment]
<a href="https://github.com/mjgiarlo"><img src="https://avatars.githubusercontent.com/u/131982?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mjgiarlo](https://github.com/mjgiarlo)**
_Saturday Oct 01, 2016 at 22:54 GMT_

----

@vantuyls I suspect Sufia should just use Blacklight's `application_name` helper, which looks in i18n though in a different location. In retrospect, we probably should never have duplicated this Blacklight functionality (using a different name) in Sufia. Here's what Blacklight does already:

https://github.com/projectblacklight/blacklight/blob/9c5d7ea48e9794960a578fe1512ed7609b634068/app/helpers/blacklight/blacklight_helper_behavior.rb#L14-L18

Basically: look in your app config for a different value at runtime, or fallback on i18n. If you find `sufia.product_name` confusing, Sufia's current impl does fallback to Blacklight's, so you could theoretically replace `sufia.product_name` with `blacklight.application_name`. We could end support for `sufia.product_name`, but that might trip people up so we could bundle that change into Sufia 8.0.0.

Thoughts, @jcoyne @jeremyf @cbeer?

**Comment by no-reply** [comment]
I'm promoting this to 2.x, since it impacts starting a new project named anything but "Hyrax".

See, e.g.: https://github.com/curationexperts/tenejo/commit/649ee9a1b4512d960623ca1c0049e5c0ff95540b

**Comment by cudevmaxwell** [comment]
Other than the locales files, `product_name` is only used in:

```
./app/models/concerns/hyrax/solr_document/export.rb:        END_NOTE_MAPPINGS.merge({ '%~' => I18n.t('hyrax.product_name'),
./app/presenters/hyrax/file_set_presenter.rb:      "#{human_readable_type} | #{title.first} | ID: #{id} | #{I18n.t('hyrax.product_name')}"
./app/presenters/hyrax/work_show_presenter.rb:      "#{human_readable_type} | #{title.first} | ID: #{id} | #{I18n.t('hyrax.product_name')}"
./app/views/shared/_footer.html.erb:        <p><%= t('hyrax.product_name') %> v<%= Hyrax::VERSION %></p>
```

`application_name` is not a internationalized field, but in practice I'm not sure `product_name` is either, and it's only used in a handful of places. Is it reasonable to just replace `product_name` with `application_name`, and call this issue closed?

(Then I need to figure out when to use `ApplicationController.helpers.application_name` and when to use `application_name` 😄)

**Comment by no-reply** [comment]
amendment to the above: `application_name` *is* i18nized as `blacklight.application_name`.

the trouble with this is that i don't see a non-breaking way out of it. anything we do to change this is going to pull the rug out from under some applications that 

> (Then I need to figure out when to use ApplicationController.helpers.application_name and when to use application_name)

it looks to me like the `ApplicationController.helpers` context is only provided in content block templates (perhaps they don't have access to the helpers otherwise? the code invoked should be the same, so if just `application_name` doesn't raise `NameError`, that's what you should use.
